test/libdnf/logger/test_loggers.cpp

```diff
@@ -62,56 +63,60 @@ void LoggersTest::test_loggers() {
+    log_router->add_logger(std::make_unique<libdnf::MemoryBufferLogger>(max_items_to_keep, reserve));
 
-    // Write messages into memory buffer loger.
+    // ====================
+    // 2. Write messages into log_router. They will be routed into memory_buffer_logger.
+    // ====================
     for (int i = 0; i < 2; ++i) {
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::CRITICAL, "Critical message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::ERROR, "Error message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::WARNING, "Warning message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::NOTICE, "Notice message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::INFO, "Info message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::DEBUG, "Debug message");
-        memory_buffer_logger->write(msg_time++, pid, libdnf::Logger::Level::TRACE, "Trace message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::CRITICAL, "Critical message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::ERROR, "Error message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::WARNING, "Warning message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::NOTICE, "Notice message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::INFO, "Info message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::DEBUG, "Debug message");
+        log_router->write(msg_time++, pid, libdnf::Logger::Level::TRACE, "Trace message");
     }
 
     // ====================
-    // 2. Create a LogRouter instance with two StreamLogger instances attached.
+    // 3. Create two StreamLogger instances which replace (swap) MemoryBufferLogger instance in log_router.
     // ====================
-    // Create log router.
-    std::unique_ptr<libdnf::LogRouter> log_router = std::make_unique<libdnf::LogRouter>();
-
     // Create two streams for logging. Save pointers to them. Pointers will be needed later in test of streams content.
     std::unique_ptr<std::ostringstream> log1_stream = std::make_unique<std::ostringstream>();
     std::unique_ptr<std::ostringstream> log2_stream = std::make_unique<std::ostringstream>();
     auto * log1_stream_ptr = log1_stream.get();
     auto * log2_stream_ptr = log2_stream.get();
 
-    // Create StreamLogger instances and attach them to LogRouter instance.
-    log_router->add_logger(std::make_unique<libdnf::StreamLogger>(std::move(log1_stream)));
+    // Create StreamLogger instance and swap it with MemoryBufferLogger instance which was added into LogRouter before.
+    std::unique_ptr<libdnf::Logger> tmp_logger = std::make_unique<libdnf::StreamLogger>(std::move(log1_stream));
+    log_router->swap_logger(tmp_logger, 0);
```

Comment:
@jrohel Please could you explain how to work with logger indexes?

Reply:
I added more descriptions to the code.